packages/components/src/components/segmented-control/segmented-control-item/segmented-control-item-styles.ts

```diff
@@ -64,32 +82,50 @@ export const getComponentCss = (
         display: 'block',
         height: '100%',
         width: '100%',
-        padding: `${pxToRemWithUnit(12)} ${ITEM_PADDING}`,
+        padding: `13px ${ITEM_PADDING}`,
         margin: 0,
-        border: 0,
+        border: `${borderWidthBase} solid ${borderColor}`,
+        transition: getTransition('border-color'),
+        outline: 0,
         background: backgroundColor,
         color: buttonColor,
         ...textSmallStyle,
-        ...getFocusJssStyle({ color: primaryColor }),
+        overflowWrap: 'normal',
+        position: 'relative',
+        borderRadius: borderRadiusSmall,
+        '&::before': {
+          content: '""',
+          position: 'absolute',
+          ...getInsetJssStyle(-5),
+          border: `${borderWidthBase} solid transparent`,
+          borderRadius: '7px',
+        },
+        '&:focus::before': {
+          borderColor: focusColor,
+        },
+        '&:focus:not(:focus-visible)::before': {
+          borderColor: 'transparent',
+        },
         ...(isDisabled
           ? {
               cursor: 'not-allowed',
             }
           : {
               cursor: 'pointer',
-              ...(!isSelected &&
-                hoverMediaQuery({
-                  transition: getTransition('background-color'),
-                  '&:hover': {
-                    background: contrastLowColor,
-                  },
-                })),
+              ...hoverMediaQuery({
+                transition: getTransition('background-color'),
```

Comment:
I think we should always apply transition to get smooth animation when switching between disabled states.